examples/chain-info/main.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	hplugin "github.com/hashicorp/go-plugin"
+
+	"chain-info/cmd"
+
+	"github.com/ignite/cli/v28/ignite/services/chain"
+	"github.com/ignite/cli/v28/ignite/services/plugin"
+)
+
+type app struct{}
+
+func (app) Manifest(_ context.Context) (*plugin.Manifest, error) {
+	return &plugin.Manifest{
+		Name:     "chain-info",
+		Commands: cmd.GetCommands(),
+	}, nil
+}
+
+func (app) Execute(ctx context.Context, c *plugin.ExecutedCommand, api plugin.ClientAPI) error {
+	// Remove the first two elements "ignite" and "flags" from OsArgs.
+	args := c.OsArgs[2:]
+
+	chainInfo, err := api.GetChainInfo(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get chain info: %w", err)
+	}
+
+	ch, err := chain.New(chainInfo.AppPath)
+	if err != nil {
+		return fmt.Errorf("failed to create a new chain object from app path: %w", err)
+	}
+
+	switch args[0] {
+	case "info":
+		return cmd.ExecuteInfo(ctx, c, ch)
+	case "build":
+		return cmd.ExecuteBuild(ctx, c, ch)
+	default:
+		return fmt.Errorf("unknown command: %s", c.Path)
+	}
+}
+
+func (app) ExecuteHookPre(_ context.Context, _ *plugin.ExecutedHook, _ plugin.ClientAPI) error {
```

Comment:
Another possibility could be:

```suggestion
func (app) ExecuteHookPre(context.Context, *plugin.ExecutedHook, plugin.ClientAPI) error {
```